irohad/ametsuchi/query_executor_visitor.hpp

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_AMETSUCHI_QUERY_EXECUTOR_VISITOR_HPP
+#define IROHA_AMETSUCHI_QUERY_EXECUTOR_VISITOR_HPP
+
+#include <memory>
+
+#include "interfaces/common_objects/types.hpp"
+#include "interfaces/permissions.hpp"
+
+namespace shared_model {
+  namespace interface {
+    class Query;
+    class QueryResponse;
+  }  // namespace interface
+}  // namespace shared_model
+
+namespace iroha {
+  namespace ametsuchi {
+
+    using QueryExecutorResult =
+        std::unique_ptr<shared_model::interface::QueryResponse>;
+
+    /**
+     * Executes query variant types
+     */
+    class QueryExecutorVisitor {
```

Comment:
It is not a `Visitor` any more. What about renaming it to `QueryExecutor`, and the one that currently holds that name - to something like `ValidatingQueryExecutor`?